libs/langchain/langchain/agents/format_scratchpad/log.py

```diff
@@ -0,0 +1,16 @@
+from typing import List, Tuple
+
+from langchain.schema.agent import AgentAction
+
+
+def format_log_to_str(
+    intermediate_steps: List[Tuple[AgentAction, str]],
+    observation_prefix: str = "Observation: ",
+    llm_prefix: str = "Thought: ",
+) -> str:
+    """Construct the scratchpad that lets the agent continue its thought process."""
+    thoughts = ""
+    for action, observation in intermediate_steps:
+        thoughts += action.log
+        thoughts += f"\n{observation_prefix}{observation}\n{llm_prefix}"
```

Comment:
imo something like this makes it easier to see what's actually happening but up to you

```suggestion
    tmpl = f"""\
{{action.log}}
{observation_prefix}{{observation}}
{llm_prefix}"""
    for action, observation in intermediate_steps:
        thoughts += tmpl.format(action=action, observation=observation)
```